fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4

```diff
@@ -158,7 +162,7 @@ MINUS_SYMBOL: '-';
 ASTERISK_SYMBOL: '*';
 SLASH_SYMBOL: '/';
 PERCENT_SYMBOL: '%';
-LOGICAL_OR: '||';
+LOGICAL_OR: '||' {setType((StarRocksParser.sqlMode & com.starrocks.qe.SqlModeHelper.MODE_PIPES_AS_CONCAT) == 0 ? LOGICAL_OR : StarRocksParser.CONCAT);};
```

Comment:
Does it means in `PIPES_AS_CONCAT` mode, all `||` expressions are recognized as `concat`, no more `LOGICAL_OR`? Is it compatible with MySQL?

Reply:
yes